tools/proteomiqon_proteininference/proteomiqon_proteininference.xml

```diff
@@ -0,0 +1,267 @@
+<tool id="proteomiqon_proteininference" name="ProteomIQon ProteinInference" version="@VERSION@" profile="20.05">
+    <description>
+        uses a list of identified peptides to deduce proteins that explain their presence in the sample.
+    </description>
+    <macros>
+        <token name="@VERSION@">0.0.7</token>
+    </macros>
+    <requirements>
+        <requirement type="package" version="@VERSION@">proteomiqon-proteininference</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        #import re
+        #if $PSMInputModeCond.PSMInputMode == "single"
+            #set basename = $re.sub(r'[^\w ,.\-+]','_',$PSMInputModeCond.scoredPSMs.element_identifier)
+            ln -s '$scoredPSMs' '${basename}.qpsm' &&
+        #elif $PSMInputModeCond.PSMInputMode == "multi"
+            #for $psmfile in $PSMInputModeCond.scoredPSMs
+                #set basename = $re.sub(r'[^\w ,.\-+]','_',$psmfile.element_identifier)
+                ln -s '$psmfile' '${basename}.qpsm' &&
+            #end for
+        #end if
+        #if $outputParamfile:
+            cat '$paramfile' >> '$out_paramfile' &&
+        #end if
+        proteomiqon-proteininference -i './' -d '$peptideDB' -p '$paramfile' -o './out'
+    ]]>
+    </command>
+    <configfiles>
+        <configfile name="paramfile">
+            <![CDATA[
+            {
+                "ProteinIdentifierRegex": "${ProteinIdentifierRegex}",
+                "Protein": {
+                  "Case": "${Protein}"
+                },
+                "Peptide": {
+                  "Case": "${Peptide}"
+                },
+                "GroupFiles": ${GroupFiles},
+                #if $GetQValueCond.GetQValue == "LogisticRegression"
+                "GetQValue": {
+                  "Case": "${GetQValueCond.GetQValue}",
+                  "Fields": [
+                    {
+                      "Case": "${GetQValueCond.LogisticRegressionType}"
+                    }
+                  ]
+                }
+                #else
+                "GetQValue": {
+                    "Case": "${GetQValueCond.GetQValue}",
+                }
+                #end if
+            }
+            ]]>
+        </configfile>
+    </configfiles>
+    <inputs>
+        <param name="peptideDB" type="data" format="sqlite" label="Peptide database" help="Specify the peptide data base."/>
+        <conditional name="PSMInputModeCond">
```

Comment:
why is this difference important?

a multiple=true datatype can also just have one?

Reply:
multiple=false allows to start batch jobs from the UI without any extra setup, I liked the possibility of being able to do that.

Reply:
Can you please explain that? You mean you would like to start N jobs for N inputs? If so what is the advantage of multiple=true?

Reply:
From my understanding, in order to use multiple files for one job, I need multiple=true. 
But if I do so, I lose the ability to start batch jobs in the way I can do it with multiple=false?
So I did this to have maximum flexibility.

To start N jobs for N inputs, I probably need to use collections, or is there another way?

Reply:
> So I did this to have maximum flexibility.

That is true.

> To start N jobs for N inputs, I probably need to use collections, or is there another way?

You don't need to, but it makes it easier. Your argumentation is completly correct. I guess what I'm asking is what is the difference for the user to:

1) run this tool on 2 datasets one ofter the other and concatenating the outputs
2) run this tool only once with 2 inputs

Is there a functional difference?

Reply:
Yes, if you run the tool with multiple files and groupFiles=true, then the tool executes the inference step using all files from all runs which can lead to peptides being included in the result which would otherwise get a bad score and get rejected.

So to put it short, the N files do have a dependency on each other when groupFiles=true.
I hope this answers your question?